components/builder-api/src/depot/server.rs

```diff
@@ -1981,16 +2024,19 @@ fn promote_package(req: &mut Request) -> IronResult<Response> {
 
     match route_message::<OriginPackagePromote, NetOk>(req, &promote) {
         Ok(_) => match route_message::<OriginPackagePromote, NetOk>(req, &promote) {
-            Ok(_) => match audit_package_rank_change(
-                req,
-                package.get_id(),
-                origin_channel.get_id(),
-                PackageChannelOperation::Promote,
-                origin_id,
-            ) {
-                Ok(_) => Ok(Response::with(status::Ok)),
-                Err(err) => return Ok(render_net_error(&err)),
-            },
+            Ok(_) => {
+                MemCache::new(config).set_package(ident.into(), &channel);
```

Comment:
putting this in the ok arm of audit_package_rank_change might result in fewer overall lines changed (== less rebase pain)